openshift/osde2e
# Validate HS MC & SC Upgrades Test Suite

# Change

This PR introduces a new test suite to run standalone (outside of osde2e) to validate hypershift management and service cluster upgrades.

Refer to the readme under the test suite folder to see further details about the test suite, how to run, etc.

Snippet from readme about the test suite:

* Before Suite
  * Validate input data
  * Identify MC/SC cluster versions (install & upgrade)
  * Get MC/SC kubeconfig's
  * Deploy ROSA Hosted Control Plane cluster to simulate customer workload
* After Suite
  * Destroy ROSA Hosted Control Plane cluster that was used for simulation
* Test Cases:
  * Perform SC upgrade
  * Verify SC post upgrade health checks
    * Runs osd-cluster-ready job
  * Verify Hosted Control Plane cluster was unaffected by the upgrade
  * Perform MC upgrade
  * Verify MC post upgrade health checks
    * Runs osd-cluster-ready job
  * Verify Hosted Control Plane cluster was unaffected by the upgrade

**Requires**

- https://github.com/openshift/osde2e-common/pull/4
- https://github.com/openshift/osde2e-common/pull/5

For [SDCICD-1028](https://issues.redhat.com//browse/SDCICD-1028)
